### ISSUE TITLE
[Accessibility]: Charts WCAG 1.4.1 Use of Color violation

### ISSUE THREAD

**Issue by kenneth-nguyen1** [carbon-design-system/carbon-charts#2054]
### Application/Team

Fusion

### Environment

Browser

### What happened? What did you expect to happen instead?

This issue was found during an accessibility audit, where colors are only being used to depict the status of the categories and legend items for a Bar Chart and Donut Chart.

There is Carbon [guidance](https://carbondesignsystem.com/data-visualization/legends/#usage) to use texture instead of color to make the chart accessible for users with visual impairment. However, there are currently no accessibility related options to achieve this when looking at the carbon-charts docs and chart configuration options.

Can this feature be explored to help make Carbon Charts accessible?

<img width="593" height="553" alt="Image" src="https://github.com/user-attachments/assets/06c6ea15-2584-4438-8ce2-087c3017dbf2" />

### What WCAG 2.1 checkpoint does the issue violate?

1.4.1 Use of Color violation

### Version

"@carbon/charts-react": "^1.19.3"

### Data & options used

```js
Example StackBlitz: https://stackblitz.com/edit/react-48bqxk4w?file=src%2Findex.js
```

### Relevant log output

```js

```

### StackBlitz repro

https://stackblitz.com/edit/react-48bqxk4w?file=src%2Findex.js

### What priority level would this issue be in your opinion?

P3
